pkg/cmd/experimental/login/login.go

```diff
@@ -36,20 +39,33 @@ prompt for user input if not provided.
 				glog.Fatalf("%v\n", err)
 			}
 
-			usernameFlag := kubecmd.GetFlagString(cmd, "username")
-			passwordFlag := kubecmd.GetFlagString(cmd, "password")
+			username := ""
 
-			accessToken, err := tokencmd.RequestToken(clientCfg, os.Stdin, usernameFlag, passwordFlag)
-			if err != nil {
-				glog.Fatalf("%v\n", err)
-			}
+			// check to see if we're already signed in.  If so, simply make sure that .kubeconfig has that information
+			if userFullName, err := whoami(clientCfg); err == nil {
```

Comment:
I *think* if the `--username` flag was provided we may want to make sure we are already signed in *with the same username specified by the flag*. That would allow users to simply switch between accounts by doing

`openshift ex login -u stark`
`openshift ex login -u targaryen`

It's already in my list btw the `osc logout` command that would also help in that flow.